packages/apollo-link-http-common/src/index.ts

```diff
@@ -44,6 +44,14 @@ export interface UriFunction {
   (operation: Operation): string;
 }
 
+// The body of a GraphQL-over-HTTP-POST request.
+export interface Body {
```

Comment:
`query` is a string rather than a DocumentNode (and there's no `context`), so I don't think that'll work?

In my experience over the entire open source and commercial Apollo codebase, guessing what type is going to be in a field or variable called `query` is the most challenging part :)